fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/MetricUnionCompositeTableDefinition.java

```diff
@@ -79,12 +84,28 @@ public MetricUnionCompositeTableDefinition(
 
     @Override
     public ConfigPhysicalTable build(ResourceDictionaries dictionaries, DataSourceMetadataService metadataService) {
+
+        // Construct a map of availability to its assigned metric
+        // by intersecting its underlying datasource metrics with table configured metrics
+        Map<Availability, Set<String>> availabilitiesToMetricNames = getPhysicalTables(dictionaries).stream()
```

Comment:
`availabilitiesToMetricNames ` is used only once. we should add a test if it's here this way, or you can inline it